src/suggestionlistworker.cpp

```diff
@@ -0,0 +1,79 @@
+#include "suggestionlistworker.h"
+#include "kiwixapp.h"
+#include <QThread>
+#include <QTimer>
+
+SuggestionListWorker::SuggestionListWorker(QObject *parent)
+: QObject(parent),
+  m_semaphore(1)
+{
+}
+
+bool SuggestionListWorker::checkAbort(const QString& text)
+{
+    QMutexLocker locker(&m_mutex);
+    if (text != m_actualSearch) {
+        emit(searchAborted());
+        m_semaphore.release();
```

Comment:
Please move the release of the resource (semaphore) in the same function/place that acquire it.